### PR TITLE
feat(text): add generating slug from words and extracting ids from slug

### PR DESCRIPTION
**Pull request by EngM7mdIbrahim**
### Screenshots

### CODE REVIEW

File: src/text.ts [hunk 1]
```diff
@@ -269,3 +269,23 @@ export function generateIDFromWord(text: string): string {
...
+}
+
+/**
+ * Extracts the id from the slug
+ * @function
+ * @param slug The slug associated with the id
+ * @returns The id from the slug
+ */
+export function extractIDfromSlug(slug: string): string | undefined {
+  return slug.split("-").pop()
```
Comment (guy-shahine) [inline src/text.ts:290]: what if someone pass a null or undefined?

File: __tests__/text.test.ts [hunk 2]
```diff
@@ -454,3 +456,25 @@ it("generateWordFromId", () => {
...
+  let expected = "hello-world-post-by-mo-qdsad123asd12dasd"
+
+  expect(actual).toBe(expected)
+
+  actual = generateSlugFromWords("qdsad123asd12dasd")
+  expected = "qdsad123asd12dasd"
+  expect(actual).toBe(expected)
+})
+
+it("extractIDfromSlug", () => {
```
Comment (guy-shahine) [inline __tests__/text.test.ts:471]: add tests for empty string, null and undefined

File: __tests__/text.test.ts [hunk 3]
```diff
@@ -454,3 +456,25 @@ it("generateWordFromId", () => {
   expected = "FinTech"
   expect(actual).toBe(expected)
 })
+
+it("generateSlugFromWords", () => {
+  let actual = generateSlugFromWords("Hello world post by mo", "qdsad123asd12dasd")
```
Comment (guy-shahine) [inline __tests__/text.test.ts:461]: no names or nicknames in our code

File: src/text.ts [hunk 4]
```diff
@@ -269,3 +269,23 @@ export function generateIDFromWord(text: string): string {
...
   return customList[id] ?? id.split("-").map(capitalizeText).join(" ")
 }
+
+/**
+ * Get the slug from words
+ * @function
+ * @param words The words that will be in the slug
+ * @returns The slug from the words
+ */
+export function generateSlugFromWords(...words: string[]): string {
```
Comment (guy-shahine) [inline src/text.ts:279]:
```suggestion
export function generateSlugFromWordsWithID(...words: string[], id: string): string {
```

File: src/text.ts [hunk 5]
```diff
@@ -269,3 +269,23 @@ export function generateIDFromWord(text: string): string {
...
+  return words.map((word) => generateIDFromWord(word)).join("-")
```
Comment (guy-shahine) [inline src/text.ts:280]: we should cover encoding unsupported characters in a url and add a comment that the slug will be url path safe

File: __tests__/text.test.ts [hunk 6]
```diff
@@ -454,3 +456,25 @@ it("generateWordFromId", () => {
...
+
+  expect(actual).toBe(expected)
+
+  actual = generateSlugFromWords("qdsad123asd12dasd")
+  expected = "qdsad123asd12dasd"
+  expect(actual).toBe(expected)
+})
+
+it("extractIDfromSlug", () => {
+  let actual = extractIDfromSlug("hello-world-post-by-mo-qdsad123asd12dasd")
```
[the remark this answers is not in the document]
Comment (EngM7mdIbrahim) [reply __tests__/text.test.ts:472]: The last word will always be the id

File: __tests__/text.test.ts (hunk 2 above)
Comment (EngM7mdIbrahim) [reply __tests__/text.test.ts:476]: addressed

File: __tests__/text.test.ts (hunk 3 above)
Comment (EngM7mdIbrahim) [reply __tests__/text.test.ts]: addressed

File: src/text.ts (hunk 4 above)
Comment (EngM7mdIbrahim) [reply src/text.ts]: addressed

File: src/text.ts (hunk 5 above)
Comment (EngM7mdIbrahim) [reply src/text.ts]: addressed

**Review by tsatsujnr139 — APPROVED** [review]

File: src/text.ts [hunk 7]
```diff
@@ -269,3 +269,32 @@ export function generateIDFromWord(text: string): string {
...
+/**
+ * Extracts the id from the slug
+ * @function
+ * @param slug The slug associated with the id
+ * @returns The id from the slug
+ */
+export function extractIDfromSlug(slug: string): string | undefined {
+  if (!slug) {
+    if (slug === "") {
+      throw new Error("slug is required, cannot be empty string")
```
Comment (guy-shahine) [inline src/text.ts:295]:
```suggestion
      throw new Error("slug cannot be empty, null or undefined string")
```

Comment (EngM7mdIbrahim) [reply src/text.ts]: addressed

**Review by guy-shahine — APPROVED** [review]
